### PR TITLE
Implement addi

### PR DESCRIPTION
**Pull request by dannywillems**

### CODE REVIEW

File: optimism/src/mips/interpreter.rs [hunk 1]
```diff
@@ -10,6 +10,10 @@ pub const FD_HINT_WRITE: u32 = 4;
 pub const FD_PREIMAGE_READ: u32 = 5;
 pub const FD_PREIMAGE_WRITE: u32 = 6;
 
+pub const REGISTER_A0: u32 = 4;
```
Comment (querolita) [inline optimism/src/mips/interpreter.rs:13]: Checked against RFC, looks okay

File: optimism/src/mips/interpreter.rs [hunk 2]
```diff
@@ -10,6 +10,10 @@ pub const FD_HINT_WRITE: u32 = 4;
...
+pub const REGISTER_A1: u32 = 5;
```
Comment (querolita) [inline optimism/src/mips/interpreter.rs:14]: Checked against RFC, looks okay

File: optimism/src/mips/interpreter.rs [hunk 3]
```diff
@@ -10,6 +10,10 @@ pub const FD_HINT_WRITE: u32 = 4;
...
+pub const REGISTER_SP: u32 = 29;
```
Comment (querolita) [inline optimism/src/mips/interpreter.rs:15]: Couldn't find where this constant comes from 😞

File: optimism/src/mips/interpreter.rs [hunk 4]
```diff
@@ -239,7 +252,18 @@ pub fn interpret_itype<Env: InterpreterEnv>(env: &mut Env, instr: ITypeInstructi
...
         ITypeInstruction::BranchGtZero => (),
-        ITypeInstruction::AddImmediate => (),
+        ITypeInstruction::AddImmediate => {
+            let rs = env.get_instruction_part(InstructionPart::RS);
+            let register_rs = env.fetch_register_checked(&rs);
+            let imm = env.get_immediate();
+            let res = register_rs + imm;
+            let rt = env.get_instruction_part(InstructionPart::RT);
+            env.overwrite_register_checked(&rt, &res);
+            env.set_instruction_pointer(env.get_instruction_pointer() + Env::constant(4u32));
```
Comment (querolita) [inline optimism/src/mips/interpreter.rs:262]: Given that this operation keeps happening over and over again, perhaps it makes sense to create a wrapper for it.

**Review by querolita — APPROVED** [review]

File: optimism/src/mips/interpreter.rs (hunk 3 above)
Comment (dannywillems) [reply optimism/src/mips/interpreter.rs:67]: See https://www.doc.ic.ac.uk/lab/secondyear/spim/node10.html

**Pull request merged**